addons/auth_signup/models/res_partner.py

```diff
@@ -97,6 +99,21 @@ def _get_signup_url_for_action(self, action=None, view_type=None, menu_id=None,
     def action_signup_prepare(self):
         return self.signup_prepare()
 
+    def get_auth_param(self):
+        """ Get a signup token related to the partner if signup is enabled.
+            If the partner already has a user, get the login parameter.
+        """
+        res = dict.fromkeys(self.ids, {})
+
+        allow_signup = safe_eval(self.env['ir.config_parameter'].get_param('auth_signup.allow_uninvited', 'False'))
```

Comment:
Ditto. Seriously, do not import safe_eval unless you have a really really strong case for it. And then, be sure to have it thoroughly reviewed, because everything matters:
 - whether methods using safe_eval are public/private
 - all parameters taken by those methods
 - ACLs protecting the value/expression that will be evaluated
 - controllers that may access the methods, and their security
 - etc..